presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -41,26 +44,41 @@
 public class HttpNativeExecutionTaskInfoFetcher
 {
     private static final Logger log = Logger.get(HttpNativeExecutionTaskInfoFetcher.class);
+    private static final String NATIVE_EXECUTION_TASK_ERROR_MESSAGE = "TaskInfoFetcher encountered too many errors talking to native process." +
+            "The process may have crashed or be under too much load.";
 
     private final PrestoSparkHttpTaskClient workerClient;
     private final ScheduledExecutorService updateScheduledExecutor;
     private final AtomicReference<TaskInfo> taskInfo = new AtomicReference<>();
     private final Executor executor;
     private final Duration infoFetchInterval;
+    private final RequestErrorTracker errorTracker;
+    private final ScheduledExecutorService errorRetryScheduledExecutor;
 
     @GuardedBy("this")
     private ScheduledFuture<?> scheduledFuture;
 
     public HttpNativeExecutionTaskInfoFetcher(
             ScheduledExecutorService updateScheduledExecutor,
+            ScheduledExecutorService errorRetryScheduledExecutor,
             PrestoSparkHttpTaskClient workerClient,
             Executor executor,
-            Duration infoFetchInterval)
+            Duration infoFetchInterval,
+            Duration maxErrorDuration)
     {
         this.workerClient = requireNonNull(workerClient, "workerClient is null");
         this.updateScheduledExecutor = requireNonNull(updateScheduledExecutor, "updateScheduledExecutor is null");
         this.executor = requireNonNull(executor, "executor is null");
         this.infoFetchInterval = requireNonNull(infoFetchInterval, "infoFetchInterval is null");
+        this.errorRetryScheduledExecutor = requireNonNull(errorRetryScheduledExecutor, "errorRetryScheduledExecutor is null");
+        this.errorTracker = new RequestErrorTracker(
```

Comment:
I see that we use RequestErrorTracker in NativeExecutionProcess. Do we need to use it in both places? Should we make sure their configuration is the same?

Reply:
I intend to let the RequestErrorTracker in TaskInfoFetcher has a shorter error out duration since in NativeExecutionProcess, it retries on the bootstrapping/Initialization of entire C++ process which is likely took longer time before the C++ process is able to response to any http call (right now the max error out duration is set to 2min). On the TaskInfoFetcher side, it retries on the periodic TaskInfoFetcher http call which currently happens every 1 second - we should tell if the C++ process crashed/lost response in much shorter duration. WDYT?

Reply:
> On the TaskInfoFetcher side, it retries on the periodic TaskInfoFetcher http call which currently happens every 1 second - we should tell if the C++ process crashed/lost response in much shorter duration

I am planning to add a NativeProcessMonitor thread to check if the process is running or not. If not running, it will throw an interrupt with appropriate error message.
The current solution of errorTracker, is a good catch all for any unhandled error scenario, but I don't think we should rely on this API call (getTaskInfo) to notify that cpp has crashed as the error message won't be accurate.

Reply:
Good call out @vermapratyush. I added one line process live check at L192 of [NativeExecutionOperator.java] (https://github.com/prestodb/presto/pull/19580/files#diff-baa3a890cdba64da929b4f4d21797c3a58c4c628b6dbb01c80133f6add5851d3) after calling the getTaskInfo which will make the error message more accurate (for the process crash).